kernel/tls_initializer/src/lib.rs

```diff
@@ -209,7 +213,7 @@ impl TlsInitializer {
         #[cfg(target_arch = "aarch64")]
         let virt_addr_value = start + self.end_of_static_sections - max(16, 8 /* TODO FIXME: pass in the TLS segment's alignment */);
 
-        section.virt_addr = VirtualAddress::new(virt_addr_value).ok_or(())?;
+        section.virt_addr = VirtualAddress::new(virt_addr_value).ok_or(TlsInitializerError::InvalidData)?;
```

Comment:
this should use an error variant related to the TLS section having an invalid virtual address (same as above)